### PR TITLE
ReadTheDocs updates for GCHP 14.5.0

### PR DESCRIPTION
**Pull request by yantosca**
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Describe the update

This PR seeks to merge documentation updates for GCHP 14.5.0 into the dev/14.5.0 branch.  We have made the following general updates:

- Moved documentation for `geoschem_config.yml`, `HEMCO_Diagn.rc`, `HEMCO_Config.rc`, `species_database.yml`, and the aerosol/chemistry/photolysis configuration files to the https://github.com/geoschem/geos-chem-shared-docs repo.
- Now point to relocated documentation files in the geos-chem-shared-docs repo
- Added usage differences between GCHP and GCC for the various documentation files
- Updated formatting to improve the page rendering and usability (i.e. using `.. option::` tags, etc)

### Related Github Issue
- https://github.com/geoschem/GCClassic/pull/68
- https://github.com/geoschem/HEMCO/pull/292

### NOTES
For now I will leave this as a draft PR as we will might have more minor updates to push.  But feel free to make comments & request changes on this PR now.  Tagging @msulprizio @lizziel @yidant

### CODE REVIEW

File: docs/source/user-guide/compiling.rst [hunk 1]
```diff
@@ -54,121 +79,158 @@ argument:
...
+   If you get a CMake error saying "Could not find XXXX" (where XXXX
+   is a dependency like ESMF, NetCDF, HDF5, etc.), the problem is that
+   CMake can't automatically find where that library is installed. You
+   can add custom paths to CMake's default search list by setting the
    :literal:`CMAKE_PREFIX_PATH` variable.
 
-   For example, if you got an error saying "Could not find ESMF", and ESMF is installed
-   to :file:`/software/ESMF`, you would do
+   For example, if you got an error saying "Could not find ESMF", and
+   ESMF is installedto :file:`/software/ESMF`, you would do
```
Comment (msulprizio) [inline docs/source/user-guide/compiling.rst]: Minor typo here. Need a space between installed and to.

File: docs/source/user-guide/config-files/ExtData_rc.rst [hunk 2]
```diff
@@ -1,57 +1,164 @@
...
-:file:`ExtData.rc` contains input variable and file read information for GCHP. 
-Explanatory information about the file is located at the top of the configuration file in all run directories. 
-The file format is the same as that used in the GEOS model, and GMAO/NASA documentation for it can be found at the ExtData component page on the GEOS-5 wiki.
-Note that this file will be retired in GCHP v15.0 when MAPL version 3 is integrated into GCHP. It will be replaced with a YAML format file with a
-simplified and easier to understand interface.
+:file:`ExtData.rc` contains input variable and file read information
+for GCHP. Explanatory information about the file is located at the top
+of the configuration file in all run directories. The file format is
+the same as that used in the GEOS model, and GMAO/NASA documentation
+for it can be found at the ExtData component page on the GEOS-5
```
Comment (msulprizio) [inline docs/source/user-guide/config-files/ExtData_rc.rst]: Should we add a link here?

**Review by msulprizio — CHANGES_REQUESTED** [review]

**Review by lizziel — APPROVED** [review]

File: docs/source/user-guide/compiling.rst (hunk 1 above)
Comment (lizziel) [reply docs/source/user-guide/compiling.rst]: Fixed.

File: docs/source/user-guide/config-files/ExtData_rc.rst (hunk 2 above)
Comment (lizziel) [reply docs/source/user-guide/config-files/ExtData_rc.rst]: Link added.

**Pull request merged**